### PR TITLE
feat(component-preview): Add component preview (part 1)

### PR DESCRIPTION
**Pull request by jpveooys**
This adds an initial version of a component to display previews of MOD.UK Frontend components.

This includes most of the required functionality. Still to implement are:

- opening examples in a new window
- displaying Nunjucks macro options

Things implemented include:

- specifying an existing component example from MOD.UK Frontend
- using the relevant code snippets automatically
- non-JavaScript version
- mobile version
- resizing the component preview

Some inspiration was taken from https://design-system.service.gov.uk, but the implementation is different (and simpler overall).

I also fixed some JavaScript errors in IE11 (which were relating to the return value of `document.querySelectorAll` not having a `forEach` method).

# Demos

# Tablet and desktop

[Screencast from 24-01-23 17:17:19.webm](https://user-images.githubusercontent.com/66470099/214362492-334f8bb2-a491-4e5e-9707-d840b54f4ad2.webm)

# Mobile

[Screencast from 24-01-23 17:16:47.webm](https://user-images.githubusercontent.com/66470099/214362519-75c83ff3-f578-4fc5-a11f-8c07e285411b.webm)

### CODE REVIEW

File: .eleventy.js [hunk 1]
```diff
@@ -53,6 +58,24 @@ module.exports = (config) => {
...
+      } catch {
+        /* Try the next include directory on error */
+      }
+    }
+    throw new Error('readFile tag: unexpectedly reached end of template include directories')
+  })
+
+  config.addShortcode(
+    'renderToString',
+    (filePath) => prettier.format(nunjucksEnv.render(filePath, {}), { parser: 'html' }),
```
Comment (jpveooys) [inline .eleventy.js:76]: The HTML generated by some of the examples is a bit messy so am using Prettier here to format it nicely.

File: e2e/tests/code-snippet.spec.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import { expect, test as base } from '@playwright/test'
```
Comment (jpveooys) [inline e2e/tests/code-snippet.spec.ts:1]: Moved these tests to a separate file (still using the back link page though).

File: e2e/tests/component-preview.spec.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import { expect, test as base } from '@playwright/test'
+
+const test = base.extend({
+  tabRole: ({ viewport }, use) => {
+    use(viewport && viewport.width < 641 ? 'button' : 'tab')
```
Comment (adamledwards) [inline e2e/tests/component-preview.spec.ts:5]: that's cool

**Review by adamledwards — APPROVED** [review]
looks good will you be add nunjunks visual regressions in part 2?

**Comment by jpveooys** [comment]
@adamledwards 

> looks good will you be add nunjunks visual regressions in part 2?

There are some there (just GitHub doesn't like rendering lots of screenshots). Do you think there are some additional ones that we should add?

**Pull request merged**